internal/envoy/debug/lsof_stat.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Tetrate
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !windows
+
+package debug
+
+import (
+	"fmt"
+	"syscall"
+)
+
+func newOpenFileStat(command string, pid int32, user string, fd uint64, name string) *openFileStat {
+	ofstat := openFileStat{
```

Comment:
I went back and forth on this in my head.. I feel like the constructor pattern feels less odd here when guarding against OS, but there is code duplication initializing the struct and that is easy to drift in the windows variant. Another way is to make a "populateNodeAndSize" function or similar and reduce the surface area.. wdyt?

Reply:
this is no longer relevant as we dumped lsof anyway